### DISCUSSION TITLE
How to update from 2018 version to 2019 version?

### DISCUSSION THREAD

**Discussion by frrobert2** [datenstrom/community#438] [Ask a question] [answered]
I was using a version of Yellow from 2018 and my default extensions were all set to txt in the config file.  I am using both the Blog and rss feed extensions.   My files are all end with txt and have html in them.

I wanted to upgrade my existing site to the latest version of Yellow. So I created a new website with the most recent version of Yellow with my existing data and changed the default extensions to txt in the config file.  The static pages display fine.  The blog pages and the rss will not display.  If I change the default extensions to md and change all my content files from txt extensions to md extensions.  The static content will display correctly.  The rss feed will display but with wrong dates due to the name change.  The Blog will only display the two sample articles but not the existing ones.

Thanks.

**Comment by schulle4u** [comment]
Have you tried updating your old installation using the [Updater](https://github.com/datenstrom/yellow-extensions/tree/master/features/update)? In most cases it should migrate all your content files. Only the layout files must be replaced/updated using a fresh repository copy.

**Comment by frrobert2** [comment]
I tried the updater and I get the following error: ERROR updating files: Can't download file 'https://github.com/datenstrom/yellow-themes/raw/master/version.ini'!

The errors from above were doing a manual migration.

**Comment by schulle4u** [comment, marked as answer]
Ah, yes, I totally forgot the removal of old API. Then we'll take the stony way. 

I would try the following: 

1. Download a fresh copy of Datenstrom Yellow and install it, followed by your desired extensions. All extensions (features formerly known as plugins, languages and themes) are now in [one repository](https://github.com/datenstrom/yellow-extensions). Please use the current available versions. 
2. Modify the file `/system/settings/system.ini` to your needs. 
3. Copy old content files to `/content` directory. 
4. I suggest to use the `.md` extension, so rename all files. 
5. Open all blog and feed pages, as well as other pages with Template settings  and change all occurrences of `Template:` to `layout:`. It is less time-consuming if you use `sed` on the command line or your text editor's search/replace in directory feature (Notepad++ can do this under windows). 

This should make your website work again. There might be other renamed settings and shortcuts, please have a closer look at individual pages. Good luck. 

HTH
Steffen

**Comment by frrobert2** [comment]
Steffen,

That worked.  I did everything but step 5 before opening the issue.  Step 5 was the key to making it work.

Thanks for your help.
